src/services/fetchTokenIntercept.js

```diff
@@ -0,0 +1,112 @@
+import _ from 'lodash';
+import Uri from 'urijs';
+import * as fetchTokenIntercept from '@shoutem/fetch-token-intercept';
+
+const AUTH_TOKENS = 'shoutem.auth.tokens';
+
+let authTokenEndpoint = null;
+let authSessionEndpoint = null;
+function initializeAuthEndpoints(authApi) {
+  authTokenEndpoint = new Uri(authApi)
+    .path('/v1/auth/tokens')
+    .toString();
+
+  authSessionEndpoint = new Uri(authApi)
+    .path('/v1/legacy-sessions')
+    .toString();
+}
+
+function createAccessTokenRequest(refreshToken) {
+  return new Request(authTokenEndpoint, {
+    headers: {
+      authorization: `Bearer ${refreshToken}`,
+      accept: 'application/vnd.api+json',
+      'content-type': 'application/vnd.api+json',
+    },
+    body: JSON.stringify({
+      data: {
+        type: AUTH_TOKENS,
+        attributes: {
+          tokenType: 'access-token',
+        },
+      },
+    }),
+    method: 'POST',
+  });
+}
+
+function shouldIntercept(baseApi) {
+  // Regex used to match urls and detect if url belongs to Shoutem api or in case of local
+  // development to locahost. Regex tries to match url string against protocol and
+  // subdomain ignoring url path. Only subdomain is tested against shoutem api base
+  // url or localhost.
+  // eslint-disable-next-line max-len
+  const apiSubdomainRegex = new RegExp(`^(https?:|ftp:)?\/\/([.a-z0-9]*?)\\.*(localhost|${baseApi})(?:\/|$|\\:)`, 'g');
```

Comment:
It would be better to only use the host of the Uri to perform this check, if possible. For example: https://medialize.github.io/URI.js/docs.html#accessors-hostname.